openfasoc/generators/glayout/glayout/flow/blocks/elementary/cascode_common_source/cascode_common_source.py

```diff
@@ -0,0 +1,407 @@
+    pdk: MappedPDK, 
+    numcols: int = 3,
+    device: str = 'nfet',
+	m1_fingers: int = 1,
+	m1_multipliers: int = 1,
+	m2_fingers: int = 1,
+	m2_multipliers: int = 1,
+    with_dummy: Optional[bool] = True,
+    with_substrate_tap: Optional[bool] = False,
+    with_tie: Optional[bool] = True,
+    tie_layers: tuple[str,str]=("met2","met1"),
+    **kwargs
+) -> Component:
+	"""An instantiable cascode common source amplifier that returns a Component object. 
+	The cascode common source amplifier could be a two transistor interdigitized structure with a shorted source and gate. 
+	It can be instantiated with either nmos or pmos devices. It can also be instantiated with a dummy device, a substrate tap, and a tie layer, and is centered at the origin. 
+	Transistor A acts as the reference and Transistor B acts as the mirror fet
+	Transistor M1 acts as the input transistor and Transistor M2 acts as the cascoded output transistor fet which determines the output impedance.
+
+	Args:
+		pdk (MappedPDK): the process design kit to use
+		numcols (int): number of columns of the interdigitized fets
+		device (str): nfet or pfet (can only interdigitize one at a time with this option)
+		m1_fingers: Number of fingers of M1 transistor
+		m1_multiplier: Number of multipliers of M1 transistor
+		with_dummy (bool): True places dummies on either side of the interdigitized fets
+		with_substrate_tap (bool): boolean to decide whether to place a substrate tapring
+		with_tie (bool): boolean to decide whether to place a tapring for tielayer
+		tie_layers (tuple[str,str], optional): the layers to use for the tie. Defaults to ("met2","met1").
+		**kwargs: The keyword arguments are passed to the two_nfet_interdigitized or two_pfet_interdigitized functions and need to be valid arguments that can be accepted by the multiplier function
+
+	Returns:
+		Component: a cascode common source amplifier component object
+	"""
+	top_level = Component("cascode common source amplifier")
+	# Create the transistors
+	print(f"Creating ", device, " devices with these parameters: ",m1_fingers,m1_multipliers, m2_fingers, m2_multipliers)
+	if device in ['nmos', 'nfet']:
+		fet_M1=nmos(pdk,
+					fingers=m1_fingers,
+					multipliers = m1_multipliers,
+					with_tie=True,
+					with_dummy=False, #with_dummy,
+					with_substrate_tap=False,
+					**kwargs)
+		fet_M2=nmos(pdk,
+					fingers=m2_fingers,
+					multipliers = m2_multipliers,
+					with_tie=True,
+					with_dummy=False, #with_dummy,
+					with_substrate_tap=False,
+					**kwargs)
+	elif device in ['pmos', 'pfet']:
+		fet_M1=pmos(pdk,
+					fingers=m1_fingers,
+					multipliers = m1_multipliers,
+					with_tie=True,
+					with_dummy=False, #with_dummy,
+					with_substrate_tap=False,
+					**kwargs)
+		fet_M2=pmos(pdk,
+					fingers=m2_fingers,
+					multipliers = m2_multipliers,
+					with_tie=True,
+					with_dummy=False, #with_dummy,
+					with_substrate_tap=False,
+					**kwargs)
+	print("FETS are instantiated now")
```

Comment:
Is this required? It might clutter the logs. Perhaps it can be changed to give out slightly more specific information.